pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go

```diff
@@ -168,3 +170,40 @@ func (dswp *desiredStateOfWorldPopulator) findAndAddActivePods() {
 	}
 
 }
+
+func isPVCReferencedInPod(pod *v1.Pod, pvc *v1.PersistentVolumeClaim) bool {
+	if pod.Namespace != pvc.Namespace {
+		return false
+	}
+	for _, podVolume := range pod.Spec.Volumes {
+		if pvcSource := podVolume.VolumeSource.PersistentVolumeClaim; pvcSource != nil && pvcSource.ClaimName == pvc.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// FindAndAddPodsByPVC finds and add active pods which reference the given
+// PVC.
+func (dswp *desiredStateOfWorldPopulator) FindAndAddActivePodsByPVC(pvc *v1.PersistentVolumeClaim) {
+	if pvc.Status.Phase != v1.ClaimBound || pvc.Spec.VolumeName == "" {
+		// Skip unbound PVCs.
+		return
+	}
+	pods, err := dswp.podLister.List(labels.Everything())
```

Comment:
Another way - perhaps we can solve this is, by implementing work queues in attach/detach controller. The problem here is - any random PVC update will cause iteration of all pods in the cluster. One can perhaps DOS the cluster by constantly updating the pvc?

Reply:
Yes, you're right. I'll update it.

Reply:
Because PVC does not contains Pod references, but we can index pods by PVC locally, then we don't need to iterate all pods every time. For example, daemon controller [indexes pods based on their NodeName](https://github.com/kubernetes/kubernetes/blob/v1.12.0-alpha.1/pkg/controller/daemon/daemon_controller.go#L197).

I think this will improve performance in large cluster.